### PR TITLE
planner: move `statsVersion` from `StatsCache` to `StatsHandler`

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #44461

Problem Summary: planner: move `statsVersion` from `StatsCache` to `StatsHandler`

### What is changed and how it works?

planner: move `statsVersion` from `StatsCache` to `StatsHandler`

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -562,14 +567,14 @@ func (h *Handle) UpdateStatsHealthyMetrics() {
 // Update reads stats meta from store and updates the stats map.
 func (h *Handle) Update(is infoschema.InfoSchema, opts ...cache.TableStatsOpt) error {
 	oldCache := h.statsCache.Load()
-	lastVersion := oldCache.Version()
+	lastVersion := h.latestUpdateTableVersion.Load()
```
Comment (hawkingrei) [inline statistics/handle/handle.go:570]: Is it better to write it like this?
```
h.statsCache.LatestUpdateTableVersion()
```

Comment (qw4990) [reply statistics/handle/handle.go:570]: No, `statsCache` should be unaware of this variable. This variable is only used when the StatsHandler reading stats data from the storage, so it's better to let the StatsHandler maintain this variable itself.

**Review by hawkingrei — APPROVED** [review]

File: statistics/handle/handle.go [hunk 2]
```diff
@@ -1010,7 +1016,22 @@ func (h *Handle) statsCacheLen() int {
...
+	}
+	for currentVer := h.latestUpdateTableVersion.Load(); maxVersion > currentVer; {
+		if h.latestUpdateTableVersion.CompareAndSwap(currentVer, maxVersion) {
+			break
+		}
+		currentVer = h.latestUpdateTableVersion.Load()
+	}
+
+	oldCache := h.statsCache.Load()
+	newCache := oldCache.CopyAndUpdate(tables, deletedIDs, opts...)
```
Comment (xuyifangreeneyes) [inline statistics/handle/handle.go:1034]: The update of version and cache pointer is not atomic. For example, both A and B call updateStatsCache, and the execution order is A.updateVersion->B.updateVersion->B.updateCache->A.updateCache.

**Pull request closed without merging**